vault/diagnose/output.go

```diff
@@ -310,62 +273,107 @@ func (t *TelemetryCollector) getOrBuildResult(id trace.SpanID) *Result {
 	return r
 }
 
+func findAttributes(e trace.Event, attr1, attr2 attribute.Key) (string, string) {
+	var av1, av2 string
+	for _, a := range e.Attributes {
+		switch a.Key {
+		case attr1:
+			av1 = a.Value.AsString()
+		case attr2:
+			av2 = a.Value.AsString()
+		}
+	}
+	return av1, av2
+}
+
 // Write outputs a human readable version of the results tree
-func (r *Result) Write(writer io.Writer) error {
+func (r *Result) Write(writer io.Writer, wrapLimit int) error {
 	var sb strings.Builder
-	r.write(&sb, 0)
+	r.write(&sb, 0, wrapLimit)
 	_, err := writer.Write([]byte(sb.String()))
 	return err
 }
 
-func (r *Result) write(sb *strings.Builder, depth int) {
+const (
+	indentString    = "  "
+	statusPrefixLen = 9
+)
+
+func indent(sb *strings.Builder, depth int) {
 	for i := 0; i < depth; i++ {
-		sb.WriteString("  ")
-	}
-	sb.WriteString(r.String())
-	sb.WriteRune('\n')
-	for _, c := range r.Children {
-		c.write(sb, depth+1)
+		sb.WriteString(indentString)
 	}
 }
 
 func (r *Result) String() string {
+	return r.StringWrapped(80)
+}
+
+func (r *Result) StringWrapped(wrapLimit int) string {
 	var sb strings.Builder
+	r.write(&sb, 0, wrapLimit)
+	return sb.String()
+}
+
+func (r *Result) write(sb *strings.Builder, depth int, limit int) {
+	indent(sb, depth)
+	var prelude string
 	if len(r.Warnings) == 0 {
 		switch r.Status {
 		case OkStatus:
-			sb.WriteString(status_ok)
+			prelude = status_ok
 		case WarningStatus:
-			sb.WriteString(status_warn)
+			prelude = status_warn
 		case ErrorStatus:
-			sb.WriteString(status_failed)
+			prelude = status_failed
 		case SkippedStatus:
-			sb.WriteString(status_skipped)
+			prelude = status_skipped
 		}
-		sb.WriteString(r.Name)
+		prelude = prelude + r.Name
 
 		if r.Message != "" || len(r.Warnings) > 0 {
```

Comment:
why do we check message or warnings, but then only include the message?

Reply:
yeah, that's not quite right.  The intent is to output the message if present, or the first warning if there are warnings but no message.  But this is outputting a blank message if there is no message but some warnings.  Will fix.